python/pip_install/parse_requirements_to_bzl/extract_single_wheel/__init__.py

```diff
@@ -29,15 +30,23 @@ def main() -> None:
     if args.extra_pip_args:
         pip_args += json.loads(args.extra_pip_args)["args"]
 
-    with NamedTemporaryFile(mode='wb') as requirement_file:
+    requirement_file = NamedTemporaryFile(mode='wb', delete=False)
+    try:
         requirement_file.write(args.requirement.encode("utf-8"))
         requirement_file.flush()
+        # Close the file so pip is allowed to read it when running on Windows.
```

Comment:
Maybe link to https://bugs.python.org/issue14243, if that's the best elaboration available.